api/src/test/java/jakarta/data/metamodel/restrict/BasicRestrictionRecordTest.java

```diff
@@ -23,109 +23,235 @@
-        BasicRestriction<Book, String> negatedTitleRestrictionBasic =
-                titleRestrictionBasic.negate();
-        BasicRestriction<Book, String> negatedNegatedTitleRestrictionBasic =
-                negatedTitleRestrictionBasic.negate();
+        BasicRestriction<Book, String> original = (BasicRestriction<Book, String>) _Book.title.equalTo("A Developer's Guide to Jakarta Data");
+        BasicRestriction<Book, String> doubleNegated = original.negate().negate();
 
         SoftAssertions.assertSoftly(soft -> {
-            soft.assertThat(titleRestrictionBasic.constraint())
-                .isEqualTo(EqualTo.value("A Developer's Guide to Jakarta Data"));
-
-            soft.assertThat(negatedNegatedTitleRestrictionBasic.constraint())
-                .isEqualTo(EqualTo.value("A Developer's Guide to Jakarta Data"));
+            soft.assertThat(original.expression()).isEqualTo(_Book.title);
+            soft.assertThat(original.constraint()).isEqualTo(EqualTo.value("A Developer's Guide to Jakarta Data"));
+            soft.assertThat(original.constraint()).isInstanceOf(EqualTo.class);
 
-            soft.assertThat(titleRestrictionBasic.constraint())
-                .isEqualTo(EqualTo.value("A Developer's Guide to Jakarta Data"));
+            soft.assertThat(doubleNegated.expression()).isEqualTo(_Book.title);
+            soft.assertThat(doubleNegated.constraint()).isEqualTo(EqualTo.value("A Developer's Guide to Jakarta Data"));
+            soft.assertThat(doubleNegated.constraint()).isInstanceOf(EqualTo.class);
         });
     }
 
     @Test
+    @DisplayName("should format toString output for greaterThan correctly")
     void shouldOutputToString() {
-        @SuppressWarnings("unchecked")
-        Restriction<Book> restriction =
-                (BasicRestriction<Book, Integer>) _Book.numPages.greaterThan(100);
+        BasicRestriction<Book, Integer> restriction = (BasicRestriction<Book, Integer>) _Book.numPages.greaterThan(100);
 
         SoftAssertions.assertSoftly(soft -> {
-            soft.assertThat(restriction.toString())
-                .isEqualTo("numPages > 100");
+            soft.assertThat(restriction.expression()).isEqualTo(_Book.numPages);
+            soft.assertThat(restriction.constraint()).isEqualTo(GreaterThan.bound(100));
+            soft.assertThat(restriction.constraint()).isInstanceOf(GreaterThan.class);
+            soft.assertThat(restriction.toString()).isEqualTo("numPages > 100");
         });
     }
 
     @Test
+    @DisplayName("should create NotEqualTo from 'notEqualTo' method directly")
     void shouldSupportNegatedRestrictionUsingDefaultConstructor() {
-        @SuppressWarnings("unchecked")
-        BasicRestriction<Book, String> negatedRestriction =
-                (BasicRestriction<Book, String>) _Book.author.notEqualTo("Unknown");
+        BasicRestriction<Book, String> restriction = (BasicRestriction<Book, String>) _Book.author.notEqualTo("Unknown");
```

Comment:
Maybe we should just use `var` here, so we can stop fighting over whether this line is too long :)

Reply:
In general, I would say the change here to merge 2 lines into a single 118 character line that developers with less acute eyesight will need to scroll right to read doesn't add any value.

Reply:
> developers with less acute eyesight

I resemble that remark.